.github/workflows/cloudformation.yml

```diff
@@ -0,0 +1,153 @@
+name: Cloudformation Actions
+
+on:
+  workflow_call:
+    inputs:
+      stack_name:
+        required: true
+        type: string
+      change_set_name:
+        required: true
+        type: string
+      template:
+        required: true
+        type: string
+      capabilities:
+        required: true
+        type: string
+      parameter_overrides:
+        required: true
+        type: string
+      execute_change_set:
+        required: true
+        type: boolean
+      target_environment:
+        required: true
+        type: string
+    secrets:
+      cf_deploy_role:
+        required: true
+
+jobs:
+  create_change_set:
+    name: Create CF Change Set
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+    - name: checkout code
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ env.BRANCH_NAME }}
+        fetch-depth: 0
+    
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        aws-region: eu-west-2
+        role-to-assume: ${{ secrets.cf_deploy_role }}
+        role-session-name: github-actions
+
+    - name: Create Change Set
+      shell: bash
+      run: |
+        aws cloudformation create-change-set \
+        --stack-name ${{ inputs.stack_name }} \
+        --change-set-name ${{ inputs.change_set_name }} \
+        --change-set-type UPDATE \
+        --template-body file://${{ inputs.template }} \
+        --capabilities ${{ inputs.capabilities }} \
+        --parameters ${{ inputs.parameter_overrides }}
```

Comment:
can you add a tags flag so resources are tagged with the version - eg https://github.com/NHSDigital/prescriptionsforpatients/blob/main/Makefile#L88
Might need to pass this through various workflows